solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetDatastreamFilter.java

```diff
@@ -60,22 +62,27 @@ public void filter(DocumentIndexingPackage dip) throws IndexingException {
         ContentObject contentObj = dip.getContentObject();
 
         FileObject fileObj = getFileObject(contentObj);
-        if (fileObj == null) {
-            return;
+        if (fileObj != null) {
+            boolean ownedByOtherObject = contentObj instanceof WorkObject;
+            // Retrieve list of datastreams associated with this object
+            List<Datastream> datastreams = getDatastreams(fileObj, ownedByOtherObject);
+            // Retrieve list of derivatives associated with the object
+            List<Datastream> derivatives = getDerivatives(fileObj.getPid(), ownedByOtherObject);
+            datastreams.addAll(derivatives);
+
+            IndexDocumentBean doc = dip.getDocument();
+
+            doc.setDatastream(getDatastreamStrings(datastreams));
+            doc.setFilesizeTotal(getFilesizeTotal(datastreams));
+            doc.setFilesizeSort(getFilesize(datastreams));
+        } else if (contentObj instanceof CollectionObject) {
```

Comment:
indexing derivatives is sufficiently generic that i think you can remove the collection object restriction